### DISCUSSION TITLE
Is there a way to skip the evaluation of the rest of the objective functions upon one objective giving a bad result?

### DISCUSSION THREAD

**Discussion by xshang93** [anyoptimization/pymoo#466] [Q&A]
Hi all, I'm getting started with Pymoo, specifically with NSGAiii and I was wondering if there is a way to skip the evaluation of rest objectives upon one functional gives bad value?

To give some context:
I have three objectives defined as separate functions, and each takes a while to evaluate. Sometimes when the first functions yields a result that is considered not feasible, I'd like to skip the rest and skip the rest, while returning a constant large value to tell the algorithm to stay away from that area.

Could anyone advise if this could  be done?

Thank you.

**Comment by blankjul** [comment]
You can use if statements during evaluation (be sure to inherit from `ElementwiseProblem`.

see the pseudo-code below:

```
import numpy as np
from pymoo.core.problem import ElementwiseProblem

class MyProblem(ElementwiseProblem):

    def __init__(self):
        super().__init__(n_var=2,
                         n_obj=2,
                         n_ieq_constr=2,
                         xl=np.array([-2,-2]),
                         xu=np.array([2,2]))

    def _evaluate(self, x, out, *args, **kwargs):
        f1 = 100 * (x[0]**2 + x[1]**2)

        if f1 > 1000:
            f2 = np.inf
        else:
            f2 = (x[0]-1)**2 + x[1]**2

        g1 = 2*(x[0]-0.1) * (x[0]-0.9) / 0.18
        g2 = - 20*(x[0]-0.4) * (x[0]-0.6) / 4.8

        out["F"] = [f1, f2]
        out["G"] = [g1, g2]
```
